secure/storage/src/on_disk.rs

```diff
@@ -71,6 +71,11 @@ impl<T: TimeService> OnDiskStorageInternal<T> {
         let contents = serde_json::to_vec(data)?;
         let mut file = File::create(self.temp_path.path())?;
         file.write_all(&contents)?;
+        // @REVIEW: many filesystems will not guarantee that the contents of the file will be stored
```

Comment:
I think that's why we have temp_path and file_path swapping

Reply:
That doesn't work... in some cases if you do:
rm a b
echo 1 > a
sync
echo 2 > b
mv b a
# crash

cat a

it's possible that you'll get a zero length file.

Reply:
This isn't production code -- but this is surprising perhaps adding fsync back in but it kills performance :P

Reply:
I think this was discussed in https://github.com/libra/libra/pull/2104/files so probably the reason we don't do fsync is perf cc: @davidiw

Reply:
Ok, if it's non-prod code would just suggest marking as such.

Reply:
It is on line 19/20 but I can add some additional emphasis.